src/coreclr/jit/llvmcodegen.cpp

```diff
@@ -1838,6 +1840,13 @@ void Llvm::buildIntegralConst(GenTreeIntConCommon* node)
 
 void Llvm::buildCall(GenTreeCall* call)
 {
+    if (isUnhandledExceptionHandler(call))
+    {
+        // Note: the exception object argument is already on the shadow stack since we're in a filter.
+        emitHelperCall(CORINFO_HELP_LLVM_EH_UNHANDLED_EXCEPTION, {}, /* doTailCall */ true);
+        return;
+    }
```

Comment:
This scheme is a hack, but one we will have to live with as we can only reference LLVM helpers in codegen.